be/src/storage/lake/transactions.cpp

```diff
@@ -373,7 +373,9 @@ StatusOr<TabletMetadataPtr> publish_version(TabletManager* tablet_mgr, int64_t t
     }
 
     // Save new metadata
-    RETURN_IF_ERROR(log_applier->finish());
+    if (log_applier != nullptr) {
```

Comment:
`log_applier` can be only nullptr if and only if `txn_offset >= txns.size()` in line 218 where the for loop never get executed.

I think it should be handled there instead of ignore the nullptr.

Reply:
@kevincai we found this issue when we run "cppcheck" on BE code.
anyways, if this code is unreachable then we can ignore the issue.